### PR TITLE
feat: update kill/restart behaviour for nested jobs 

### PR DESCRIPTION
**Pull request by abyesilyurt**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/tests/conftest.py [hunk 1]
```diff
@@ -156,48 +156,6 @@ def low_worker() -> Worker:
     del worker
 
 
-@pytest.fixture(scope="function")
```
Comment (abyesilyurt) [inline packages/syft/tests/conftest.py:160]: unused.

File: tests/integration/local/twin_api_sync_test.py [hunk 2]
```diff
@@ -48,48 +45,6 @@ def get_ds_client(client: DomainClient) -> DomainClient:
     return client.login(email="a@a.com", password="asdf")
 
 
-@pytest.fixture(scope="function")
```
Comment (abyesilyurt) [inline tests/integration/local/twin_api_sync_test.py:51]: duplicate, already in conftest

File: packages/syft/src/syft/service/job/job_service.py [hunk 3]
```diff
@@ -158,28 +189,60 @@ def update(
...
+        # return SyftSuccess(message="Job killed successfully!")
+
+        wait_until(lambda: job.fetched_status == JobStatus.INTERRUPTED)
+        wait_until(
+            lambda: all(
+                subjob.fetched_status == JobStatus.INTERRUPTED for subjob in job.subjobs
+            )
+        )
+
+        return SyftSuccess(message="Job killed successfully!")
```
Comment (abyesilyurt) [inline packages/syft/src/syft/service/job/job_service.py]: any suggestions on error handling here?

File: packages/syft/src/syft/service/job/job_service.py [hunk 4]
```diff
@@ -158,28 +189,60 @@ def update(
         res = res.ok()
         return SyftSuccess(message="Great Success!")
 
+    def _kill(self, context: AuthedServiceContext, job: Job) -> SyftSuccess | SyftError:
```
Comment (teo-milea) [inline packages/syft/src/syft/service/job/job_service.py:192]: took me a second to realise the actual termination of the job is happening in the queue in the monitor function, could we add some comments to highlight that?

File: packages/syft/src/syft/service/job/job_service.py (hunk 3 above)
Comment (teo-milea) [reply packages/syft/src/syft/service/job/job_service.py]: I think the commented approach is fine for the moment, if need we can improve it later

**Review by teo-milea — APPROVED** [review]
Great job!!

File: packages/syft/src/syft/service/job/job_service.py (hunk 4 above)
Comment (abyesilyurt) [reply packages/syft/src/syft/service/job/job_service.py:192]: done

**Pull request merged**